server/cmd/mmctl/commands/compliance_export.go

```diff
@@ -75,3 +88,72 @@ func complianceExportCancelCmdF(c client.Client, command *cobra.Command, args []
 
 	return nil
 }
+
+func complianceExportDownloadCmdF(c client.Client, command *cobra.Command, args []string) error {
+	jobID := args[0]
+	var path string
+	if len(args) > 1 {
+		path = args[1]
+	} else {
+		path = jobID + ".zip"
+	}
+
+	retries, _ := command.Flags().GetInt("num-retries")
+
+	var outFile *os.File
+	info, err := os.Stat(path)
+	switch {
+	case err != nil && !os.IsNotExist(err):
+		// some error occurred and not because file doesn't exist
+		return fmt.Errorf("failed to stat compliance export file: %w", err)
+	case err == nil && info.Size() > 0:
+		// we exit to avoid overwriting an existing non-empty file
+		return fmt.Errorf("compliance export file already exists")
+	case err != nil:
+		// file does not exist, we create it
+		outFile, err = os.Create(path)
+	default:
+		// no error, file exists, we open it
+		outFile, err = os.OpenFile(path, os.O_WRONLY, 0600)
+	}
+
+	if err != nil {
+		return fmt.Errorf("failed to create/open compliance export file: %w", err)
+	}
+	defer outFile.Close()
+
+	i := 0
+	var suggestedFilename string
+	for i < retries+1 {
+		suggestedFilename, err = c.DownloadComplianceExport(context.TODO(), jobID, outFile)
+		if err != nil {
+			if i == retries {
+				return fmt.Errorf("failed to download compliance export file: %w", err)
+			}
+			i++
+			fmt.Printf("Download attempt %d/%d failed: %v. Retrying...\n", i, retries+1, err)
+			continue
+		}
+		break
+	}
```

Comment:
Another nit/personal preference: can't we use a normal loop here? Something like
```go
	for i := 0; i < retries + 1; i++ {
		suggestedFilename, err = c.DownloadComplianceExport(context.TODO(), jobID, outFile)
		if err == nil {
			break
		}
		
		if i == retries {
			return fmt.Errorf("failed to download compliance export file: %w", err)
		}

		fmt.Printf("Download attempt %d/%d failed: %v. Retrying...\n", i, retries+1, err)
	}
```

Reply:
Yep, good point, fixed.